ResearchKit/Common/UIApplication+ResearchKit.m

```diff
@@ -34,15 +34,20 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 
 @implementation UIApplication (ResearchKit)
 
-- (UIResponder *)ork_currentFirstResponder {
+- (UIView *)ork_currentFirstResponderView {
     NSMutableArray *sender = [NSMutableArray new] ;
     [self sendAction:@selector(ork_updateCurrentFirstResponder:)
                                                to:nil
                                              from:sender
                                          forEvent:nil];
     
+    for (UIResponder *responder in sender) {
+        if ([responder isKindOfClass:[UIView class]]) {
+            return (UIView *)responder;
+        }
```

Comment:
We don't need to walk this array - the array only has one object. (see implementation of ork_updateCurrentFirstResponder:)

Why not just do:
id responder = [sender firstObject];
if (! [responder isKindOfClass:[UIView class]]) {
    responder = nil;
}
return responder;


This also would neatly avoid an early return.